yjit/src/backend/arm64/mod.rs

```diff
@@ -146,12 +146,13 @@ impl Assembler
             }
         }
 
-        self.forward_pass(|asm, index, op, opnds, target, text, pos_marker, original_opnds| {
-            // Load all Value operands into registers that aren't already a part
-            // of Load instructions.
-            let opnds = match op {
-                Op::Load => opnds,
-                _ => opnds.into_iter().map(|opnd| {
+        let mut asm = Assembler::new_with_label_names(std::mem::take(&mut self.label_names));
```

Comment:
I think you might be able to get rid of the `&mut asm` borrows added in the diff by doing:

```suggestion
        let mut asm = Assembler::new_with_label_names(std::mem::take(&mut self.label_names));
        let asm = &mut asm;
```
Or maybe you need to pick a different name for the owning local.

Reply:
That might cause borrow checker errors actually because that makes a long mutable borrow. Don't worry about it if that doesn't work. Though maybe we should consider turning those functions that take a `&mut Assembler` into methods so we don't need to type `&mut` as often.

Reply:
It works, I've updated